### PR TITLE
Download feature bundles from bintray

### PR DESCRIPTION
**Pull request by markusjura**
The `sandbox run` now retrieves all bundles from bintray, not only the monitoring feature. Bootstrapping the features in the ConductR [docker-entrypoiny.py](https://github.com/typesafehub/conductr/blob/master/conductr/src/universal/bin/docker-entrypoint.py) is not necessary anymore. The docker-entrypoint script for ConductR 2.0 has been changed accordingly: https://github.com/typesafehub/conductr/pull/1531

The following bundles are now downloaded from bintray:
- conductr-haproxy + conductr-haproxy-dev-mode configuration
- eslite
- conductr-elasticsearch
- conductr-kibana
- cinnamon-grafana
- visualizer

The new logic is only applied from ConductR 2.0.0+ only because the bundles hosted on bintray are only compatible with this ConductR version.

When ConductR 1.1.x- is used, then the old logic of bootstrapping the bundles in the docker-entrypoint.py script is used.

Both has been tested with manual integration tests locally. Unit tests have been added for the new cases as well.

The visualization and logging feature has been changed so that the bundles are now getting downloaded from bintray, in the same fashion as the monitoring feature. Another benefit is that the `sandbox run` command now shows the progress of the feature startup and waits until the feature related bundles have been started in ConductR.

Also, this PR does a couple of other minor changes:
- Introducing a `--disable-instructions` flag for all conduct commands, to suppress log messages at the end the command that instructs the user how to load, unload a bundle or get the current bundle status. This flag is used when the conduct commands are used inside `sandbox run` command to not print unnecessary instructions even when the run command has not finished yet. At the end of the `sandbox run` command, the user will get a summary and further instructions anyway
- Making the `sandbox run` output more readable by grouping the steps and giving each group a headline (see PR sample output below)
- At the end of the `sandbox run` command the current bundle status is printed (see PR sample output below). This is only applied for ConductR 2.0.0+ and is now possible because we ensure beforehand that all bundles have been started

### CODE REVIEW

**Comment by markusjura** [comment]
Unintentionally closed

**Comment by markusjura** [comment]
**Output: sandbox run**

```
sandbox run 0.1.0 --feature monitoring --feature visualization
|------------------------------------------------|
| Stopping ConductR                              |
|------------------------------------------------|
cond-0
|------------------------------------------------|
| Starting ConductR                              |
|------------------------------------------------|
Starting container cond-0 exposing 127.0.0.1:3000, 127.0.0.1:5601, 127.0.0.1:9200, 127.0.0.1:9999..
b6894c944347d067904db670171f6e36a99ed684ecfb2942a657da0f84d9177c
Waiting for ConductR to start..
|------------------------------------------------|
| Starting HAProxy                               |
|------------------------------------------------|
Deploying bundle conductr-haproxy with configuration conductr-haproxy-dev-mode
Retrieving bundle..
Loading bundle from cache typesafe/bundle/conductr-haproxy
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving file:///Users/mj/workspace/conductr-cli/conductr-haproxy
Resolving bundle typesafe/bundle/conductr-haproxy
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving https://dl.bintray.com/typesafe/bundle/typesafe/conductr-haproxy/v2-316bb3cfd249eea3c7da7d8b8108f41b6c815ca6a49268f24e21dca2d236e792/conductr-haproxy-v2-316bb3cfd249eea3c7da7d8b8108f41b6c815ca6a49268f24e21dca2d236e792.zip
[#################################################] 100%
Retrieving configuration..
Loading bundle configuration from cache typesafe/bundle-configuration/conductr-haproxy-dev-mode
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving file:///Users/mj/workspace/conductr-cli/conductr-haproxy-dev-mode
Resolving bundle configuration typesafe/bundle-configuration/conductr-haproxy-dev-mode
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving https://dl.bintray.com/typesafe/bundle-configuration/typesafe/conductr-haproxy-dev-mode/v2-d86c4eb272a4d75d86156983b23fde8b149393395c2dab0e8ea24ba862702691/conductr-haproxy-dev-mode-v2-d86c4eb272a4d75d86156983b23fde8b149393395c2dab0e8ea24ba862702691.zip
[#################################################] 100%
Loading bundle to ConductR..
[#################################################] 100%
Bundle 316bb3cfd249eea3c7da7d8b8108f41b-d86c4eb272a4d75d86156983b23fde8b waiting to be installed
Bundle 316bb3cfd249eea3c7da7d8b8108f41b-d86c4eb272a4d75d86156983b23fde8b installed
Bundle loaded.
Bundle run request sent.
Bundle 316bb3cfd249eea3c7da7d8b8108f41b-d86c4eb272a4d75d86156983b23fde8b waiting to reach expected scale 1
Bundle 316bb3cfd249eea3c7da7d8b8108f41b-d86c4eb272a4d75d86156983b23fde8b has scale 0, expected 1.
Bundle 316bb3cfd249eea3c7da7d8b8108f41b-d86c4eb272a4d75d86156983b23fde8b expected scale 1 is met
|------------------------------------------------|
| Starting logging feature based on              |
| elasticsearch and kibana                       |
|------------------------------------------------|
Deploying bundle conductr-elasticsearch..
Retrieving bundle..
Loading bundle from cache typesafe/bundle/conductr-elasticsearch
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving file:///Users/mj/workspace/conductr-cli/conductr-elasticsearch
Resolving bundle typesafe/bundle/conductr-elasticsearch
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving https://dl.bintray.com/typesafe/bundle/typesafe/conductr-elasticsearch/v1-5c55487ce6ad537dc7adac45c994182d480e8c5e7331c46c98f95eec6bf55777/conductr-elasticsearch-v1-5c55487ce6ad537dc7adac45c994182d480e8c5e7331c46c98f95eec6bf55777.zip
[#################################################] 100%
Loading bundle to ConductR..
[#################################################] 100%
Bundle 5c55487ce6ad537dc7adac45c994182d waiting to be installed
Bundle 5c55487ce6ad537dc7adac45c994182d installed
Bundle loaded.
Bundle run request sent.
Bundle 5c55487ce6ad537dc7adac45c994182d waiting to reach expected scale 1
Bundle 5c55487ce6ad537dc7adac45c994182d has scale 0, expected 1..
Bundle 5c55487ce6ad537dc7adac45c994182d expected scale 1 is met
Deploying bundle conductr-kibana..
Retrieving bundle..
Loading bundle from cache typesafe/bundle/conductr-kibana
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving file:///Users/mj/workspace/conductr-cli/conductr-kibana
Resolving bundle typesafe/bundle/conductr-kibana
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving https://dl.bintray.com/typesafe/bundle/typesafe/conductr-kibana/v1-bff3cdcd5d35a085540b1a47824040a2aeedccfc948247b7a402b44a80b80c79/conductr-kibana-v1-bff3cdcd5d35a085540b1a47824040a2aeedccfc948247b7a402b44a80b80c79.zip
[#################################################] 100%
Loading bundle to ConductR..
[#################################################] 100%
Bundle bff3cdcd5d35a085540b1a47824040a2 waiting to be installed
Bundle bff3cdcd5d35a085540b1a47824040a2 installed
Bundle loaded.
Bundle run request sent.
Bundle bff3cdcd5d35a085540b1a47824040a2 waiting to reach expected scale 1
Bundle bff3cdcd5d35a085540b1a47824040a2 has scale 0, expected 1.
Bundle bff3cdcd5d35a085540b1a47824040a2 expected scale 1 is met
|------------------------------------------------|
| Starting monitoring feature                    |
|------------------------------------------------|
Deploying bundle cinnamon-grafana..
Retrieving bundle..
Loading bundle from cache typesafe/bundle/cinnamon-grafana
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving file:///Users/mj/workspace/conductr-cli/cinnamon-grafana
Resolving bundle typesafe/bundle/cinnamon-grafana
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving https://dl.bintray.com/typesafe/bundle/typesafe/cinnamon-grafana/v2-385f2b0e798f151635c5bbc8c623fac2b13390c53ea977cbf05163db4e4a10af/cinnamon-grafana-v2-385f2b0e798f151635c5bbc8c623fac2b13390c53ea977cbf05163db4e4a10af.zip
[#################################################] 100%
Loading bundle to ConductR..
[#################################################] 100%
Bundle 385f2b0e798f151635c5bbc8c623fac2 waiting to be installed
Bundle 385f2b0e798f151635c5bbc8c623fac2 installed
Bundle loaded.
Bundle run request sent.
Bundle 385f2b0e798f151635c5bbc8c623fac2 waiting to reach expected scale 1
Bundle 385f2b0e798f151635c5bbc8c623fac2 has scale 0, expected 1.....
Bundle 385f2b0e798f151635c5bbc8c623fac2 expected scale 1 is met
|------------------------------------------------|
| Starting visualization feature                 |
|------------------------------------------------|
Deploying bundle visualizer..
Retrieving bundle..
Loading bundle from cache typesafe/bundle/visualizer
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving file:///Users/mj/workspace/conductr-cli/visualizer
Resolving bundle typesafe/bundle/visualizer
Bintray credentials loaded from /Users/mj/.bintray/.credentials
Retrieving https://dl.bintray.com/typesafe/bundle/typesafe/visualizer/v2-7fe30fb02ff5df19b09fa33cfe03c5cfa4eb7fb4ff8d7422b9662074eca8b333/visualizer-v2-7fe30fb02ff5df19b09fa33cfe03c5cfa4eb7fb4ff8d7422b9662074eca8b333.zip
[#################################################] 100%
Loading bundle to ConductR..
[#################################################] 100%
Bundle 7fe30fb02ff5df19b09fa33cfe03c5cf waiting to be installed
Bundle 7fe30fb02ff5df19b09fa33cfe03c5cf installed
Bundle loaded.
Bundle run request sent.
Bundle 7fe30fb02ff5df19b09fa33cfe03c5cf waiting to reach expected scale 1
Bundle 7fe30fb02ff5df19b09fa33cfe03c5cf has scale 0, expected 1...
Bundle 7fe30fb02ff5df19b09fa33cfe03c5cf expected scale 1 is met
|------------------------------------------------|
| Summary                                        |
|------------------------------------------------|
ConductR has been started
Check resource consumption of Docker container that run the ConductR node with:
  docker stats cond-0
Check current bundle status with:
  conduct info
Bundle status:
ID               NAME                    #REP  #STR  #RUN
385f2b0          cinnamon-grafana           1     0     1
bff3cdc          conductr-kibana            1     0     1
5c55487          conductr-elasticsearch     1     0     1
7fe30fb          visualizer                 1     0     1
316bb3c-d86c4eb  conductr-haproxy           1     0     1
```

File: conductr_cli/conduct_main.py [hunk 1]
```diff
@@ -217,7 +227,7 @@ def build_parser(dcos_mode):
                             help='The optional number of executions, defaults to 1')
     run_parser.add_argument('--affinity',
                             default=None,
-                            help='The optional ID of the bundle to run alongside with (v1.1 onwards)')
+                            help='The optional ID of the bundle to run alongside with (v2.0 onwards)')
```
Comment (fsat) [inline conductr_cli/conduct_main.py:230]: Thanks for picking this up @markusjura

**Review by fsat — APPROVED** [review]

File: conductr_cli/sandbox_common.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
                   9005,  # ConductR controlServer
                   9006}  # ConductR bundleStreamServer
 CONDUCTR_DEV_IMAGE = 'typesafe-docker-registry-for-subscribers-only.bintray.io/conductr/conductr'
-LATEST_CONDUCTR_VERSION = '1.1.9'
+LATEST_CONDUCTR_VERSION = '1.1.11'
```
Comment (huntc) [inline conductr_cli/sandbox_common.py:10]: Unsure why this is needed at all..

**Review by huntc — APPROVED** [review]

**Pull request merged**